contk/_utils/file_utils.py

```diff
@@ -0,0 +1,173 @@
+"""
+Utilities for working with the local dataset cache.
+This file is adapted from the AllenNLP library at https://github.com/allenai/allennlp
+Copyright by the AllenNLP authors.
+"""
+
+import os
+import logging
+import json
+import tempfile
+import shutil
+import hashlib
+
+from tqdm import tqdm
+from checksumdir import dirhash
+
+import requests
+
+class Downloader():
+	r'''Base class of Downloader
+	'''
+	def __init__(self):
+		logging.basicConfig(level=logging.INFO, \
+				format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+		self.logger = logging.getLogger(__name__)
+
+		self.dataset_cache_path = os.path.join(os.getenv('HOME'), 'dataset_cache')
+		self.config_dir = '.'
+		os.makedirs(self.dataset_cache_path, exist_ok=True)
+
+	def get_config(self, res_name):
+		r'''Get config(dict) by the name of resource
+		'''
+		config_path = os.path.join(self.config_dir, res_name + '.json')
+		if not os.path.exists(config_path):
+			raise FileNotFoundError("file {} not found".format(config_path))
+		with open(config_path, 'r') as config_file:
+			config = json.load(config_file)
+
+		return config
+
+
+	def http_get(self, url, temp_file):
+		"""Pull a file directly from http"""
+		req = requests.get(url, stream=True)
+		content_length = req.headers.get('Content-Length')
+		total = int(content_length) if content_length is not None else None
+		progress = tqdm(unit="B", total=total)
+		for chunk in req.iter_content(chunk_size=1024):
+			if chunk:
+				progress.update(len(chunk))
+				temp_file.write(chunk)
+		progress.close()
+
+
+	def preprocess(self, local_path):
```

Comment:
preprocess and postprocess is different by different resource type, so maybe it's better to make this two functions be a class